### PR TITLE
TT-5683 use paratext fobai permission to allow publish

### PR DESCRIPTION
**Pull request by sarahentzel**

### CODE REVIEW

File: src/utils/useCanPublish.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  const dispatch = useDispatch();
+  const getCanPublish = (token: string, errorReporter: any, message: string) =>
+    dispatch(action.getCanPublish(token, errorReporter, message));
+  const resetCanPublish = () => dispatch(action.resetCanPublish());
+  const t: IProfileStrings = useSelector(profileSelector, shallowEqual);
+
+  useEffect(() => {
+    if (!isOffline) {
+      if (!askingRef.current && accessToken && !paratext_canPublishStatus) {
+        askingRef.current = true; //so we only call it once
```
Comment (gtryus) [inline src/utils/useCanPublish.ts:30]: Where do we reset this askingRef?

File: src/components/Sheet/PlanSheet.tsx [hunk 2]
```diff
@@ -695,7 +695,7 @@ export function PlanSheet(props: IProps) {
   };
 
   const handlePublishToggle: MouseEventHandler<HTMLButtonElement> = () => {
-    if (!hasParatext) {
+    if (!canPublish && hidePublishing) {
```
Comment (gtryus) [inline src/components/Sheet/PlanSheet.tsx:698]: Why does hidePublishing have to be true here?

**Review by gtryus — CHANGES_REQUESTED** [review]
I made the Jira number TT-5683 which should be moved back from Ready for Testing when Jira is available again.

I had two questions for you.

File: src/utils/useCanPublish.ts (hunk 1 above)
Comment (sarahentzel) [reply src/utils/useCanPublish.ts:30]: why would we need to call it again?

Comment (gtryus) [reply src/utils/useCanPublish.ts:30]: perhaps only if it failed because of a bad connection?

**Pull request merged**